Adventure implementation breaks on 1.20.2 because of 1.20.4

**Expected behaviour:**
The commands are executed on 1.20.2 Paper as well because it is backwards supported.

**Actual behaviour:**
When an user uses the commands on 1.20.2 Paper, the following error is going to happen and it can't execute the commands

```
Caused by: java.lang.NoSuchMethodError: 'java.util.List net.kyori.adventure.text.TranslatableComponent.arguments()'
	at net.onelitefeather.antiredstoneclockremastered.translations.PluginTranslationRegistry.translate(PluginTranslationRegistry.java:60) ~[AntiRedstoneClock-Remastered-1.0.1-all.jar:?]
	at net.kyori.adventure.translation.GlobalTranslatorImpl.translate(GlobalTranslatorImpl.java:90) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer$1.renderTranslatable(TranslatableComponentRenderer.java:82) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer$1.renderTranslatable(TranslatableComponentRenderer.java:74) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.AbstractComponentRenderer.render(AbstractComponentRenderer.java:50) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.event.HoverEvent$Action$1.render(HoverEvent.java:814) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.event.HoverEvent$Action$1.render(HoverEvent.java:811) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.event.HoverEvent.withRenderedValue(HoverEvent.java:289) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.mergeStyle(TranslatableComponentRenderer.java:238) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.mergeStyleAndOptionallyDeepRender(TranslatableComponentRenderer.java:222) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.renderText(TranslatableComponentRenderer.java:169) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.AbstractComponentRenderer.render(AbstractComponentRenderer.java:48) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.lambda$optionallyRenderChildrenAppendAndBuild$0(TranslatableComponentRenderer.java:228) ~[adventure-api-4.14.0.jar:?]
	at java.util.ArrayList.forEach(ArrayList.java:1511) ~[?:?]
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1092) ~[?:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.optionallyRenderChildrenAppendAndBuild(TranslatableComponentRenderer.java:228) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.mergeStyleAndOptionallyDeepRender(TranslatableComponentRenderer.java:223) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.TranslatableComponentRenderer.renderText(TranslatableComponentRenderer.java:169) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.text.renderer.AbstractComponentRenderer.render(AbstractComponentRenderer.java:48) ~[adventure-api-4.14.0.jar:?]
	at net.kyori.adventure.translation.GlobalTranslator.render(GlobalTranslator.java:87) ~[adventure-api-4.14.0.jar:?]
	at io.papermc.paper.adventure.PaperAdventure.translated(PaperAdventure.java:207) ~[paper-1.20.2.jar:git-Paper-228]
	at io.papermc.paper.adventure.PaperAdventure.asJsonString(PaperAdventure.java:192) ~[paper-1.20.2.jar:git-Paper-228]
	at net.minecraft.network.FriendlyByteBuf.writeComponent(FriendlyByteBuf.java:533) ~[?:?]
	at net.minecraft.network.protocol.game.ClientboundSystemChatPacket.write(ClientboundSystemChatPacket.java:46) ~[paper-1.20.2.jar:git-Paper-228]
	at net.minecraft.network.PacketEncoder.encode(PacketEncoder.java:42) ~[paper-1.20.2.jar:git-Paper-228]
	... 77 more


```
**Version:** AntiRedstoneClock-Remastered-1.0.1-all

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -51,7 +51,9 @@ repositories {
 dependencies {
     compileOnly(libs.paper)
     implementation(libs.bstats)
-    implementation(libs.adventure)
+    implementation(libs.adventure.api)
+    implementation(libs.minimessage)
+    implementation(libs.adventure.bukkit)
     implementation(libs.cloud.command.paper)
     implementation(libs.cloud.command.extras)
     implementation(libs.cloud.command.annotations)
@@ -87,7 +89,7 @@ tasks {
     }
     shadowJar {
         relocate("org.bstats", "net.onelitefeather.antiredstoneclockremastered.org.bstats")
-
+        relocate("net.kyori", "net.onelitefeather.antiredstoneclockremastered.net.kyori")
     }
 }
 

--- a/build.gradle.kts
+++ b/build.gradle.kts
@@ -51,6 +51,7 @@ repositories {
 dependencies {
     compileOnly(libs.paper)
     implementation(libs.bstats)
+    implementation(libs.adventure)
     implementation(libs.cloud.command.paper)
     implementation(libs.cloud.command.extras)
     implementation(libs.cloud.command.annotations)
@@ -86,6 +87,7 @@ tasks {
     }
     shadowJar {
         relocate("org.bstats", "net.onelitefeather.antiredstoneclockremastered.org.bstats")
+
     }
 }
 

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -27,9 +27,12 @@ dependencyResolutionManagement {
             version("bstats", "3.0.2")
             version("cloudcommand", "2.0.0-SNAPSHOT")
             version("adventure", "4.16.0")
+            version("adventure.platform", "4.3.2")
 
             library("paper", "io.papermc.paper", "paper-api").versionRef("paper")
-            library("adventure", "net.kyori", "adventure-api").versionRef("adventure")
+            library("adventure.api", "net.kyori", "adventure-api").versionRef("adventure")
+            library("minimessage", "net.kyori", "adventure-text-minimessage").versionRef("adventure")
+            library("adventure.bukkit", "net.kyori", "adventure-platform-bukkit").versionRef("adventure.platform")
             library("bstats", "org.bstats", "bstats-bukkit").versionRef("bstats")
             library("cloud.command.paper", "org.incendo", "cloud-paper").versionRef("cloudcommand")
             library("cloud.command.annotations", "org.incendo", "cloud-annotations").versionRef("cloudcommand")

--- a/settings.gradle.kts
+++ b/settings.gradle.kts
@@ -26,8 +26,10 @@ dependencyResolutionManagement {
             version("paper", "1.20.4-R0.1-SNAPSHOT")
             version("bstats", "3.0.2")
             version("cloudcommand", "2.0.0-SNAPSHOT")
+            version("adventure", "4.16.0")
 
             library("paper", "io.papermc.paper", "paper-api").versionRef("paper")
+            library("adventure", "net.kyori", "adventure-api").versionRef("adventure")
             library("bstats", "org.bstats", "bstats-bukkit").versionRef("bstats")
             library("cloud.command.paper", "org.incendo", "cloud-paper").versionRef("cloudcommand")
             library("cloud.command.annotations", "org.incendo", "cloud-annotations").versionRef("cloudcommand")

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
@@ -140,9 +140,10 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
         }
         annotationParser = new AnnotationParser<>(commandManager, CommandSender.class);
         annotationParser.descriptionMapper(string -> RichDescription.of(Component.translatable(string)));
-        MinecraftHelp<CommandSender> help = MinecraftHelp.createNative(
+        MinecraftHelp<CommandSender> help = MinecraftHelp.create(
                 "/arcm help",
-                commandManager
+                commandManager,
+                sender -> this.adventure.sender(sender)
         );
         commandManager.command(
                 commandManager.commandBuilder("arcm").literal("help")

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
@@ -69,7 +69,7 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
         return this.adventure;
     }
 
-    public final static Component PREFIX = MiniMessage.miniMessage().deserialize("<gradient:red:white>[AntiRedstoneClock]</gradient>");
+    public static final Component PREFIX = MiniMessage.miniMessage().deserialize("<gradient:red:white>[AntiRedstoneClock]</gradient>");
 
     @Override
     public void onLoad() {

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/AntiRedstoneClockRemastered.java
@@ -1,6 +1,7 @@
 package net.onelitefeather.antiredstoneclockremastered;
 
 import net.kyori.adventure.key.Key;
+import net.kyori.adventure.platform.bukkit.BukkitAudiences;
 import net.kyori.adventure.text.Component;
 import net.kyori.adventure.text.minimessage.MiniMessage;
 import net.kyori.adventure.translation.GlobalTranslator;
@@ -28,10 +29,10 @@ import org.bukkit.Material;
 import org.bukkit.command.CommandSender;
 import org.bukkit.plugin.Plugin;
 import org.bukkit.plugin.java.JavaPlugin;
+import org.checkerframework.checker.nullness.qual.NonNull;
 import org.incendo.cloud.annotations.AnnotationParser;
 import org.incendo.cloud.bukkit.CloudBukkitCapabilities;
 import org.incendo.cloud.component.DefaultValue;
-import org.incendo.cloud.description.Description;
 import org.incendo.cloud.execution.ExecutionCoordinator;
 import org.incendo.cloud.minecraft.extras.MinecraftHelp;
 import org.incendo.cloud.minecraft.extras.RichDescription;
@@ -59,6 +60,14 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
 
     private Metrics metrics;
     private AnnotationParser<CommandSender> annotationParser;
+    private BukkitAudiences adventure;
+
+    public @NonNull BukkitAudiences adventure() {
+        if(this.adventure == null) {
+            throw new IllegalStateException("Tried to access Adventure when the plugin was disabled!");
+        }
+        return this.adventure;
+    }
 
     public final static Component PREFIX = MiniMessage.miniMessage().deserialize("<gradient:red:white>[AntiRedstoneClock]</gradient>");
 
@@ -72,6 +81,7 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
 
     @Override
     public void onEnable() {
+        this.adventure = BukkitAudiences.create(this);
         final TranslationRegistry translationRegistry = new PluginTranslationRegistry(TranslationRegistry.create(Key.key("antiredstoneclockremastered", "translations")));
         translationRegistry.defaultLocale(Locale.US);
         Path langFolder = getDataFolder().toPath().resolve("lang");
@@ -98,6 +108,14 @@ public final class AntiRedstoneClockRemastered extends JavaPlugin {
         enableBStatsSupport();
         registerEvents();
         registerCommands();
+    }
+
+    @Override
+    public void onDisable() {
+        if(this.adventure != null) {
+            this.adventure.close();
+            this.adventure = null;
+        }
     }
 
     private void registerCommands() {

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/DisplayActiveClocksCommand.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/DisplayActiveClocksCommand.java
@@ -23,8 +23,8 @@ public final class DisplayActiveClocksCommand {
     @CommandDescription("antiredstoneclockremastered.command.display.description")
     @Permission("antiredstoneclockremastered.command.display")
     public void displayClocks(CommandSender commandSender) {
-        commandSender.sendMessage(Component.translatable("antiredstoneclockremastered.command.display.clock.title").arguments(AntiRedstoneClockRemastered.PREFIX));
-        this.plugin.getRedstoneClockService().getRedstoneClocks().stream().map(this::mapClockToMessage).forEach(commandSender::sendMessage);
+        this.plugin.adventure().sender(commandSender).sendMessage(Component.translatable("antiredstoneclockremastered.command.display.clock.title").arguments(AntiRedstoneClockRemastered.PREFIX));
+        this.plugin.getRedstoneClockService().getRedstoneClocks().stream().map(this::mapClockToMessage).forEach(component -> this.plugin.adventure().sender(commandSender).sendMessage(component));
 
     }
 

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/FeatureCommand.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/FeatureCommand.java
@@ -73,9 +73,9 @@ public final class FeatureCommand {
 
     private void sendMessageToggleMessage(CommandSender sender, boolean value) {
         if (value) {
-            sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.toggle.enabled").arguments(AntiRedstoneClockRemastered.PREFIX));
+            this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.toggle.enabled").arguments(AntiRedstoneClockRemastered.PREFIX));
         } else {
-            sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.toggle.disabled").arguments(AntiRedstoneClockRemastered.PREFIX));
+            this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.toggle.disabled").arguments(AntiRedstoneClockRemastered.PREFIX));
         }
     }
 
@@ -88,7 +88,7 @@ public final class FeatureCommand {
         plugin.getConfig().set("check.ignoredWorlds", worlds);
         plugin.saveConfig();
         plugin.getRedstoneClockService().reload();
-        sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.world.add").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(world.getName())));
+        this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.world.add").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(world.getName())));
     }
 
     @Command("check ignored_worlds remove <world>")
@@ -100,7 +100,7 @@ public final class FeatureCommand {
         plugin.getConfig().set("check.ignoredWorlds", worlds);
         plugin.saveConfig();
         plugin.getRedstoneClockService().reload();
-        sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.world.remove").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(world.getName())));
+        this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.world.remove").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(world.getName())));
     }
 
     @Command("check ignored_regions add <region>")
@@ -112,7 +112,7 @@ public final class FeatureCommand {
         plugin.getConfig().set("check.ignoredRegions", regions);
         plugin.saveConfig();
         plugin.getRedstoneClockService().reload();
-        sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.region.add").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(region)));
+        this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.region.add").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(region)));
     }
 
     @Command("check ignored_regions remove <region>")
@@ -124,7 +124,7 @@ public final class FeatureCommand {
         plugin.getConfig().set("check.ignoredRegions", regions);
         plugin.saveConfig();
         plugin.getRedstoneClockService().reload();
-        sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.region.remove").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(region)));
+        this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.check.region.remove").arguments(AntiRedstoneClockRemastered.PREFIX, Component.text(region)));
     }
 
     @Command("clock autoBreak")
@@ -188,7 +188,7 @@ public final class FeatureCommand {
     }
 
     private void sendMessageSetMessage(CommandSender sender, Integer value) {
-        sender.sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.clock.set").arguments(AntiRedstoneClockRemastered.PREFIX, TranslationArgument.numeric(value)));
+        this.plugin.adventure().sender(sender).sendMessage(Component.translatable("antiredstoneclockremastered.command.feature.clock.set").arguments(AntiRedstoneClockRemastered.PREFIX, TranslationArgument.numeric(value)));
     }
 
 }

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/ReloadCommand.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/commands/ReloadCommand.java
@@ -20,7 +20,7 @@ public final class ReloadCommand {
     @Permission("antiredstoneclockremastered.command.reload")
     public void reloadConfig(CommandSender commandSender) {
         this.plugin.getRedstoneClockService().reload();
-        commandSender.sendMessage(Component.translatable("antiredstoneclockremastered.command.reload.success").arguments(AntiRedstoneClockRemastered.PREFIX));
+        plugin.adventure().sender(commandSender).sendMessage(Component.translatable("antiredstoneclockremastered.command.reload.success").arguments(AntiRedstoneClockRemastered.PREFIX));
 
     }
 }

--- a/src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java
+++ b/src/main/java/net/onelitefeather/antiredstoneclockremastered/service/RedstoneClockService.java
@@ -163,7 +163,7 @@ public final class RedstoneClockService {
                                 executor.teleport(location);
                             }
                         })));
-        player.sendMessage(component);
+        this.antiRedstoneClockRemastered.adventure().player(player).sendMessage(component);
     }
 
     private void breakBlock(@NotNull Location location) {

